masonite/drivers/QueueAsyncDriver.py

```diff
@@ -19,22 +23,60 @@ def __init__(self, app: App):
         """
         self.container = app
 
-    def push(self, *objects, args=(), callback='handle', ran=1, channel=None):
+    def _threading(self):
+        """ Implements Async by Threading """
+
+        # Necessary for Python 3.4
+        if self.workers is None:
+            # Use this number because ThreadPoolExecutor is often
+            # used to overlap I/O instead of CPU work.
+            self.workers = (os.cpu_count() or 1) * 5
+        if self.workers <= 0:
+            raise QueueException("max_workers must be greater than 0")
+
+        with ThreadPoolExecutor(max_workers=self.workers) as executor:
+            for obj in self.objects:
+                if inspect.isclass(obj):
+                    obj = self.container.resolve(obj)
+
+                try:
+                    executor.submit(
+                        fn=getattr(obj, self.callback), args=self.args, kwargs=self.kwargs)
+                except AttributeError:
+                    # Could be wanting to call only a method asyncronously
+                    executor.submit(fn=obj, args=self.args, kwargs=self.kwargs)
+
+    def _multiprocessing(self):
+        """ Implements Async by Multiprocesses """
+
+        with ProcessPoolExecutor(max_workers=self.workers) as executor:
+            for obj in self.objects:
+                if inspect.isclass(obj):
+                    obj = self.container.resolve(obj)
+
+                try:
+                    executor.submit(
+                        fn=getattr(obj, self.callback), args=self.args, kwargs=self.kwargs)
+                except AttributeError:
+                    # Could be wanting to call only a method asyncronously
+                    executor.submit(fn=obj, args=self.args, kwargs=self.kwargs)
+
```

Comment:
We can break the 2 methods above into a single method because the only difference is that one line. So we can make that a variable of `self.processor` for example and we can set that in the push method or more preferably make it a special `self._get_processor()` method since we may need to set some special cases like for Python 3.4 and threading